homeassistant/components/hassio/backup.py

```diff
@@ -360,8 +364,16 @@ async def async_restore_backup(
         restore_homeassistant: bool,
     ) -> None:
         """Restore a backup."""
-        if restore_homeassistant and not restore_database:
-            raise HomeAssistantError("Cannot restore Home Assistant without database")
+        manager = self._hass.data[DATA_MANAGER]
+        # The backup manager has already checked that the backup exists so we don't need to
+        # check that here.
+        backup = await manager.backup_agents[agent_id].async_get_backup(backup_id)
+        if (
+            backup
+            and restore_homeassistant
+            and restore_database != backup.database_included
```

Comment:
Hm, this implements what you write here:

> It's not possible to restore database from a backup which doesn't have database or not restore the database from a backup which has database

and from what I understand what Supervisor is capable of today, correct?

IMHO, not restoring the data should be possible, but there is the concern what if this is a downgrade of Home Assistant Core, and the (currently newer) database is not compatible? :thinking:  The Core will complain on startup, so it is "safe", but the outcome might not be what the user expect (since he presumably wants to continue with the current database). 

When no database is part of the backup, the user can already run into this situation, so we are a bit inconsistent in what we allow today :cry: